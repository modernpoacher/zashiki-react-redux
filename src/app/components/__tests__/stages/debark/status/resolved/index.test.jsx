import React from 'react'
import PropTypes from 'prop-types'

import {
  toSnapshot
} from 'react-component-snapshot'

import '@testing-library/jest-dom'

import {
  render
} from '@testing-library/react'

import Resolved from '#zashiki-react-redux/app/components/stages/debark/status/resolved'

const MOCK_STRING_DEFINITION = {
  type: 'STRING',
  params: {
    answer: {
      title: 'String',
      value: 'string'
    },
    changeAnswer: {
      href: '/alpha/omega',
      text: 'Change',
      visuallyHiddenText: 'string'
    }
  }
}

const MOCK_NUMBER_DEFINITION = {
  type: 'NUMBER',
  params: {
    answer: {
      title: 'Number',
      value: '3'
    },
    changeAnswer: {
      href: '/alpha/omega',
      text: 'Change',
      visuallyHiddenText: 'number'
    }
  }
}

const MOCK_BOOLEAN_DEFINITION = {
  type: 'BOOLEAN',
  params: {
    answer: {
      title: 'Boolean',
      value: 'false'
    },
    changeAnswer: {
      href: '/alpha/omega',
      text: 'Change',
      visuallyHiddenText: 'boolean'
    }
  }
}

const MOCK_NULL_DEFINITION = {
  type: 'NULL',
  params: {
    answer: {
      title: 'Null',
      value: 'null'
    },
    changeAnswer: {
      href: '/alpha/omega',
      text: 'Change',
      visuallyHiddenText: 'null'
    }
  }
}

const MOCK_RESOURCE = {
  alpha: 'MOCK ALPHA',
  omega: 'MOCK OMEGA'
}

const MOCK_RESPONSE = {
  '#/': 'MOCK VALUE'
}

const MOCK_ERRORS = []

const MOCK_DEFINITIONS = [
  {
    definition: [
      MOCK_STRING_DEFINITION,
      MOCK_NUMBER_DEFINITION,
      MOCK_BOOLEAN_DEFINITION,
      MOCK_NULL_DEFINITION
    ],
    resource: MOCK_RESOURCE,
    response: MOCK_RESPONSE,
    errors: MOCK_ERRORS
  }
]

/**
 *  @param {{ to: string | { pathname: string }, children: React.ReactNode | React.ReactNode[] }} props
 *  @returns {React.JSX.Element}
 */
function MockLink ({ to, children }) {
  if (typeof to === 'string') {
    return (
      <a href={to} className='mock-link'>
        {children}
      </a>
    )
  }

  const {
    pathname
  } = to

  return (
    <a href={pathname} className='mock-link'>
      {children}
    </a>
  )
}

MockLink.propTypes = {
  to: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.shape({})
  ]),
  children: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.arrayOf(
      PropTypes.node
    )
  ])
}

jest.mock('react-router', () => {
  return {
    __esModule: true,
    Link: MockLink
  }
})

describe('#zashiki-react-redux/app/components/stages/debark/status/resolved', () => {
  describe('Always', () => {
    it('renders', () => {
      expect(toSnapshot(render(
        <Resolved
          definitions={MOCK_DEFINITIONS}
          token={{}}
          onSubmit={jest.fn()}
        />
      )))
        .toMatchSnapshot()
    })
  })
})
