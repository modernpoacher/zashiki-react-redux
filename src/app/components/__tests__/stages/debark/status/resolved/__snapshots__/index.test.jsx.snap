// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#zashiki-react-redux/app/components/stages/debark/status/resolved Always renders 1`] = `
<div
  className="debark resolved"
>
  <Debark>
    <form
      action="/debark-stage"
      method="POST"
      onSubmit={[Function]}
    >
      <Fragment>
        <CheckAnswersSprocket>
          <div
            className="sprocket check-answers"
          >
            <CheckAnswersTitle />
            <CheckAnswersGroup>
              <dl
                className="group check-answers"
              >
                <div
                  className="answer"
                >
                  <AnswerTitle>
                    <dt
                      className="answer-title"
                    >
                      String
                    </dt>
                  </AnswerTitle>
                  <AnswerValue>
                    <dd
                      className="answer-value"
                    >
                      string
                    </dd>
                  </AnswerValue>
                  <ChangeAnswer>
                    <dd
                      className="change-answer"
                    >
                      <MockLink>
                        <a
                          className="mock-link"
                          href="/alpha/omega"
                        >
                          <LinkText>
                            <TextContentWithVisuallyHiddenText>
                              <TextContent>
                                <span
                                  className="text-content"
                                >
                                  Change
                                </span>
                              </TextContent>
                               
                              <VisuallyHiddenText>
                                <span
                                  className="visually-hidden-text"
                                >
                                  string
                                </span>
                              </VisuallyHiddenText>
                            </TextContentWithVisuallyHiddenText>
                          </LinkText>
                        </a>
                      </MockLink>
                    </dd>
                  </ChangeAnswer>
                </div>
                <div
                  className="answer"
                >
                  <AnswerTitle>
                    <dt
                      className="answer-title"
                    >
                      Number
                    </dt>
                  </AnswerTitle>
                  <AnswerValue>
                    <dd
                      className="answer-value"
                    >
                      3
                    </dd>
                  </AnswerValue>
                  <ChangeAnswer>
                    <dd
                      className="change-answer"
                    >
                      <MockLink>
                        <a
                          className="mock-link"
                          href="/alpha/omega"
                        >
                          <LinkText>
                            <TextContentWithVisuallyHiddenText>
                              <TextContent>
                                <span
                                  className="text-content"
                                >
                                  Change
                                </span>
                              </TextContent>
                               
                              <VisuallyHiddenText>
                                <span
                                  className="visually-hidden-text"
                                >
                                  number
                                </span>
                              </VisuallyHiddenText>
                            </TextContentWithVisuallyHiddenText>
                          </LinkText>
                        </a>
                      </MockLink>
                    </dd>
                  </ChangeAnswer>
                </div>
                <div
                  className="answer"
                >
                  <AnswerTitle>
                    <dt
                      className="answer-title"
                    >
                      Boolean
                    </dt>
                  </AnswerTitle>
                  <AnswerValue>
                    <dd
                      className="answer-value"
                    >
                      false
                    </dd>
                  </AnswerValue>
                  <ChangeAnswer>
                    <dd
                      className="change-answer"
                    >
                      <MockLink>
                        <a
                          className="mock-link"
                          href="/alpha/omega"
                        >
                          <LinkText>
                            <TextContentWithVisuallyHiddenText>
                              <TextContent>
                                <span
                                  className="text-content"
                                >
                                  Change
                                </span>
                              </TextContent>
                               
                              <VisuallyHiddenText>
                                <span
                                  className="visually-hidden-text"
                                >
                                  boolean
                                </span>
                              </VisuallyHiddenText>
                            </TextContentWithVisuallyHiddenText>
                          </LinkText>
                        </a>
                      </MockLink>
                    </dd>
                  </ChangeAnswer>
                </div>
                <div
                  className="answer"
                >
                  <AnswerTitle>
                    <dt
                      className="answer-title"
                    >
                      Null
                    </dt>
                  </AnswerTitle>
                  <AnswerValue>
                    <dd
                      className="answer-value"
                    >
                      null
                    </dd>
                  </AnswerValue>
                  <ChangeAnswer>
                    <dd
                      className="change-answer"
                    >
                      <MockLink>
                        <a
                          className="mock-link"
                          href="/alpha/omega"
                        >
                          <LinkText>
                            <TextContentWithVisuallyHiddenText>
                              <TextContent>
                                <span
                                  className="text-content"
                                >
                                  Change
                                </span>
                              </TextContent>
                               
                              <VisuallyHiddenText>
                                <span
                                  className="visually-hidden-text"
                                >
                                  null
                                </span>
                              </VisuallyHiddenText>
                            </TextContentWithVisuallyHiddenText>
                          </LinkText>
                        </a>
                      </MockLink>
                    </dd>
                  </ChangeAnswer>
                </div>
              </dl>
            </CheckAnswersGroup>
          </div>
        </CheckAnswersSprocket>
      </Fragment>
      <Token />
      <button
        type="submit"
      >
        Continue
      </button>
    </form>
  </Debark>
</div>
`;
